stablehlo/dialect/StablehloOps.td

```diff
@@ -2871,7 +2871,9 @@ def StableHLO_TransposeOp: StableHLO_ShapedInterfaceOp<"transpose",
     HLO_TensorOrPerAxisQuantizedTensor:$operand,
     DenseI64ArrayAttr:$permutation
   );
-  let results = (outs HLO_TensorOrPerAxisQuantizedTensor);
+  let results = (outs HLO_TensorOrPerAxisQuantizedTensor:$result);
+
+  let hasVerifier = 1;
```

Comment:
cc @ghpvnist - I think the new verification code should go in `TypeInference.cpp > inferTransposeOp`, per https://github.com/openxla/stablehlo/blob/main/docs/type_inference.md#p3-maintain-verification-code-in-verifiers-and-shape-functions

Does this line up with your expectations having written many of these?

Reply:
This setup is similar to `ReverseOp` (and probably a few others) but I'm not sure if that's out of model as well

Reply:
At a minimum lets update the verification doc linked above

Reply:
`+ AddOp` 
what we conclude here is also applicable for AddOp

Reply:
thanks @ghpvnist  missed your comment.. maybe didn't refreshed the page. 

`HLO_CompatibleOperandsAndResultElementTyp`e doesn't cover all of `transpose_c1` 
also `transpose_c4` is not covered.

Reply:
Okay makes sense. The point of labeling `HLO_CompatibleOperandsAndResultElementType` with `transpose_c1` is that it tests at least a subset of c1. We're good as long as we aren't writing tests that ODS already covers.

Reply:
As the result typeof transpose is fully inferable, it can be created suing `TransposeOp::build` which does not take result type as input and hence it makes sense to  move the verification logic of transpose to inferTransposeOp to avoid missing on any verification during op creation.

For ops like quantized version of AddOp, the result type cannot be inferred as the scales and zero-points can vary. In order to create such an op, we need to use a version of AddOP::build which takes result type as parameter and that builder will never invoke `Add::inferReturnTypes`. So thereis no point for AddOp verfication logic for quantization to move to type ingerence. 

WIth that it makes sense to move the code from verification to type inference if
1. The op has region.
2. The type inference is feasible.

Note that the reason reshapeOp and broadcast_in_dim op has separete verifioers is because their type cannot be inferred. 

We should update https://github.com/openxla/stablehlo/blob/main/docs/type_inference.md#p3-maintain-verification-code-in-verifiers-and-shape-functions with this information for future use.

Reply:
thanks folks. 

* updated `type_inference.md` at https://github.com/openxla/stablehlo/pull/2158 as conclusion from this thread. 
* `TransposeOp` : since return type inference is feasible, will move the verification logic from custom verifier to `inferTransposeOp`
* `AddOp` : no change needed because return type inference is not feasible for quantized types. 
* I will also take a relook at existing ops and need of dedicated verifier.

reopening this comment for further  discussion if needed..